public/docs/ts/latest/guide/dependency-injection.jade

```diff
@@ -748,105 +717,117 @@ code-example(format, language="html").
-  +makeExample('dependency-injection/ts/app/providers.component.ts','providers-9a-interface')(format=".")
-  +makeExample('dependency-injection/ts/app/providers.component.ts','provider-9a-ctor-interface')(format=".")
-  :marked
-    That seems strange if we're used to dependency injection in strongly typed languages, where
-    an interface is the preferred dependency lookup key.
-
-    It's not Angular's fault. An interface is a TypeScript design-time artifact. JavaScript doesn't have interfaces.
-    The TypeScript interface disappears from the generated JavaScript.
-    There is no interface type information left for Angular to find at runtime.
-// end Typescript only
-
-// #docregion tokens-opaque-1
-<a id="opaque-token"></a>
-- var lang = current.path[1]
-- var opaquetoken = lang == 'dart' ? '<code>OpaqueToken</code>' : '<a href="../api/core/OpaqueToken-class.html"><code>OpaqueToken</code></a>'
-h3 OpaqueToken
-p.
   The solution is to define and use an !{opaquetoken}.
   The definition looks like this:
-// #enddocregion tokens-opaque-1
+
 +makeExample('dependency-injection/ts/app/app.config.ts','token')(format='.')
+
 :marked
   We register the dependency provider using the `OpaqueToken` object:
-+makeExample('dependency-injection/ts/app/providers.component.ts','providers-9b')(format=".")
-// #docregion tokens-opaque-2
-- var lang = current.path[1]
-- var decorated = lang == 'dart' ? 'annotated' : 'decorated'
-- var configuration = lang == 'dart' ? '' : 'configuration'
-:marked
-  Now we can inject the #{configuration} object into any constructor that needs it, with
-  the help of an `@Inject` #{decorator} that tells Angular how to find the #{configuration} dependency value.
-// #enddocregion tokens-opaque-2
-+makeExample('dependency-injection/ts/app/providers.component.ts','provider-9b-ctor')(format=".")
-
-// begin Typescript only
+
++makeExample('dependency-injection/ts/app/providers.component.ts','providers-9')(format=".")
+
+:marked
+  Now we can inject the configuration object into any constructor that needs it, with
+  the help of an `@Inject` #{_decorator}:
+
++makeExample('dependency-injection/ts/app/app.component.2.ts','ctor')(format=".")
+
+- var configType = _docsFor == 'dart' ? '<code>Map</code>' : '<code>AppConfig</code>'
 .l-sub-section
   :marked
     Although it plays no role in dependency injection,
-    the `Config` interface supports strong typing of the configuration object within the class.
+    the !{configType} interface supports typing of the configuration object within the class.
+
+block dart-map-alternative
+  :marked
+    Or we can provide and inject the configuration object in our top-level `AppComponent`.
+
+  +makeExcerpt('app/app.component.ts','providers')
+
+#optional
 :marked
-// end typescript only
+  ## Optional dependencies
+
+  Our `HeroService` *requires* a `Logger`, but what if it could get by without
+  a logger?
+  As is illustrated next, we can indicate that to the dependency injection
```

Comment:
This sounds very formal. Also, we should mention `@Optional()` in the text, not just in the code. Maybe:

We can tell Angular that the dependency is optional by annotating the constructor argument with `@Optional()`: